### PR TITLE
Customer Home: Add better copy for completed email and mobile app tasks

### LINKED ISSUE #39416: Customer Home: Completed checklist details are confusing.
When looking at the completed checklist, we're offered additional details in the card expansion. For the two I've seen, they're quite confusing; they seem to imply you still have to do (or should do again) the action that is marked as done. We should either give more appropriate next actions for the, uh, next action, or update the text to better reflect the completed context.

Screen Shot 2020-02-12 at 9 56 05 AM
Screen Shot 2020-02-12 at 9 56 37 AM

### PR DESCRIPTION
**Pull request by sixhours**
#### Changes proposed in this Pull Request

* The original mobile app and email verification completed copy was confusing, suggesting the task had not actually been completed. Let's update it to make that clearer.
* /cc @obi2020 and others for feedback/suggestions, my copy in this PR is probably not rock solid. Suggestions welcome!

**Before**

Screen Shot 2020-02-18 at 1 25 55 PM
Screen Shot 2020-02-18 at 1 25 48 PM

**After**

Screen Shot 2020-02-18 at 1 18 04 PM
Screen Shot 2020-02-19 at 8 53 54 AM

#### Testing instructions

* Switch to this PR, create a new site, and navigate to Customer Home
* Complete email verification and downloading the mobile app
* You should see the tasks in the completed list, with updated copy

Fixes #39416

### CODE REVIEW

File: client/my-sites/checklist/wpcom-checklist/component.jsx [hunk 1]
```diff
@@ -601,8 +612,9 @@ class WpcomChecklistComponent extends PureComponent {
 			<TaskComponent
 				{ ...baseProps }
 				bannerImageSrc="/calypso/images/stats/tasks/mobile-app.svg"
-				completedButtonText={ translate( 'Download mobile app' ) }
+				completedButtonText={ translate( 'Re-download mobile app' ) }
```
Comment (a8ci18n) [inline client/my-sites/checklist/wpcom-checklist/component.jsx]: :warning: This change will be queued for retranslation. We'll display the English text until then.

**Comment by obi2020** [comment]
Copy LGTM... except how would the email get validated if there's a typo?

**Comment by gwwar** [comment]
> except how would the email get validated if there's a typo?

Good point @obi2020. The only case I can think of is an email alias. Maybe the prompt can instead be around managing, or switching the main account adress?

**Comment by gwwar** [comment]
As an aside, string translations get picked up on Mondays I think so if we tidy up copy on Fridays and apply the string freeze label, we should get the quickest turn around on landing copy changes with i18n translations.

**Comment by sixhours** [comment]
> except how would the email get validated if there's a typo?

> Good point @obi2020. The only case I can think of is an email alias. Maybe the prompt can instead be around managing, or switching the main account adress?

Good thinking, I've updated the copy to ask if they need to change something.

**Pull request merged**